core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp

```diff
@@ -1377,20 +1550,120 @@ KOKKOS_INLINE_FUNCTION void parallel_reduce(
     const Impl::TeamVectorRangeBoundariesStruct<
         iType, Impl::OpenMPTargetExecTeamMember>& loop_boundaries,
     const Lambda& lambda, ValueType& result) {
-  ValueType* tmp_scratch =
+  static_assert(sizeof(ValueType) <=
+                Impl::OpenMPTargetExecTeamMember::TEAM_REDUCE_SIZE);
```

Comment:
Is this a FIXME?

Reply:
Can't this be implemented in terms of the reducer variant using `Kokkos::Sum`?

Reply:
> Is this a FIXME?

No this is not a FIXME, essentially we are restricting the nested array-reduction size to be <= 32 elements per team. 
The reduction uses a global scratch, so 32 elements with 16 bytes per element is essentially 512 bytes per team. Anything more than this might be  detrimental for the performance. 



> Can't this be implemented in terms of the reducer variant using `Kokkos::Sum`?

We were trying to rely on OpenMP constructs as much as possible.

Reply:
Maybe I am misunderstanding but does that mean that you can't do reductions for large types (and that is never intended to work)?

Reply:
I want to say its better to not perform reductions larger than 32 elements as long as we have the current implementation strategy where we allocate 16 bytes per element in the scratch memory. But maybe @crtrott can answer this better.

Reply:
May question solely is if we should put a FIXME there or not. :slightly_smiling_face:

Reply:
Yes you are right. We should put a FIXME there. Will do it.

Reply:
Actually I forgot that we have already added a FIXME_OPENMPTARGET for this issue in one of the variants of the `parallel_reduce` functions. I will repeat that everywhere.

Reply:
This `static_assert` seems to be duplicated now looking at lines 1573-1577.